tests/testAutopas/tests/selectors/tuningStrategy/PredictiveTuningTest.cpp

```diff
@@ -0,0 +1,584 @@
+  EXPECT_TRUE(predictiveTuning.searchSpaceIsTrivial());
+  EXPECT_THAT(predictiveTuning.getAllowedContainerOptions(),
+              ::testing::ElementsAre(autopas::ContainerOption::directSum));
+}
+
+TEST_F(PredictiveTuningTest, testSearchSpaceMoreOptions) {
+  unsigned int iteration = 0;
+  autopas::PredictiveTuning predictiveTuning(
+      {autopas::ContainerOption::linkedCells}, {1.}, {autopas::TraversalOption::c08}, {autopas::DataLayoutOption::soa},
+      {autopas::Newton3Option::enabled, autopas::Newton3Option::disabled}, iteration);
+  EXPECT_FALSE(predictiveTuning.searchSpaceIsEmpty());
+  EXPECT_FALSE(predictiveTuning.searchSpaceIsTrivial());
+  EXPECT_THAT(predictiveTuning.getAllowedContainerOptions(),
+              ::testing::ElementsAre(autopas::ContainerOption::linkedCells));
+}
+
+TEST_F(PredictiveTuningTest, testRemoveN3OptionRemoveAll) {
+  unsigned int iteration = 0;
+  autopas::PredictiveTuning predictiveTuning(
+      {autopas::ContainerOption::linkedCells}, {1.}, {autopas::TraversalOption::c08, autopas::TraversalOption::sliced},
+      {autopas::DataLayoutOption::soa, autopas::DataLayoutOption::aos}, {autopas::Newton3Option::enabled}, iteration);
+
+  EXPECT_THROW(predictiveTuning.removeN3Option(autopas::Newton3Option::enabled),
+               autopas::utils::ExceptionHandler::AutoPasException);
+}
+
+TEST_F(PredictiveTuningTest, testRemoveN3OptionRemoveSome) {
+  unsigned int iteration = 0;
+  autopas::PredictiveTuning predictiveTuning(
+      {autopas::ContainerOption::linkedCells}, {1.}, {autopas::TraversalOption::c08, autopas::TraversalOption::sliced},
+      {autopas::DataLayoutOption::soa, autopas::DataLayoutOption::aos},
+      {autopas::Newton3Option::enabled, autopas::Newton3Option::disabled}, iteration);
+
+  EXPECT_NO_THROW(predictiveTuning.removeN3Option(autopas::Newton3Option::enabled));
+  EXPECT_FALSE(predictiveTuning.searchSpaceIsEmpty());
+  EXPECT_FALSE(predictiveTuning.searchSpaceIsTrivial());
+}
+
+/*
+ * Tests the prediction and the selection of the right optimum.
+ * Three different configurations:
+ *      - C08 that is not optimal but it getting getting less expensive.
+ *      - Sliced is optimal in the beginning but is getting more expensive.
+ *      - C01 is constant an not in the optimum range.
+ * In the third iteration c08 should be predicted to be the optimum.
+ */
+TEST_F(PredictiveTuningTest, testSelectPossibleConfigurations) {
+  unsigned int iteration = 0;
+  autopas::PredictiveTuning predictiveTuning(
+      {autopas::ContainerOption::linkedCells}, {1.},
+      {autopas::TraversalOption::c08, autopas::TraversalOption::c01, autopas::TraversalOption::sliced},
+      {autopas::DataLayoutOption::soa}, {autopas::Newton3Option::disabled}, iteration);
+
+  predictiveTuning.reset();
+
+  EXPECT_EQ(autopas::Configuration(autopas::ContainerOption::linkedCells, 1., autopas::TraversalOption::c08,
+                                   autopas::DataLayoutOption::soa, autopas::Newton3Option::disabled),
```

Comment:
for readability, you should probably create variables for all the configurations that appear here that have concise names pointing out their differences. E.g.
```C++
const auto configurationC08 = autopas::Configuration(autopas::ContainerOption::linkedCells, 
                                                     1.,
                                                     autopas::TraversalOption::c08, 
                                                     autopas::DataLayoutOption::soa, 
                                                     autopas::Newton3Option::disabled);
const auto configurationSliced = ...
```